Framework/MPIAlgorithms/src/BroadcastWorkspace.cpp

```diff
@@ -91,9 +92,7 @@ void BroadcastWorkspace::exec() {
   for (std::size_t i = 0; i < numSpec; ++i) {
     if (world.rank() == root) {
       // For local output, just copy over
-      outputWorkspace->dataX(i) = inputWorkspace->readX(i);
-      outputWorkspace->dataY(i) = inputWorkspace->readY(i);
-      outputWorkspace->dataE(i) = inputWorkspace->readE(i);
+      outputWorkspace->setHistogram(i, inputWorkspace->histogram(i));
 
       // Send out the current spectrum
       broadcast(world, const_cast<MantidVec &>(inputWorkspace->readX(i)), root);
```

Comment:
Sorry, I didn't copy it from the issue description, BroadcastWorkspace and GatherWorkspaces need more work, as the `boost::mpi::broadcast(...)` function does not work with HistogramData types, and serialisation needs to be implemented. I've created an issue for that #18717, which will also cover the HistogramData refactoring